google/cloud/aiplatform/prediction/handler.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+try:
+    from fastapi import Request
+    from fastapi import Response
+except ImportError:
+    raise ImportError(
+        "FastAPI is not installed and is required to build model servers. "
+        'Please install the SDK using "pip install python-aiplatform[prediction]"'
+    )
+
+from google.cloud.aiplatform.prediction.predictor import Predictor
+from google.cloud.aiplatform.prediction.serializer import DefaultSerializer
+
+
+class Handler:
+    """Interface for Handler class to handle prediction requests."""
+
+    def __init__(self, predictor: Predictor):
```

Comment:
Let's actually not require the user provide a Predictor (they may not necessarily use the `Predictor` interface.

We'll need to pass the gcs_artifacts_uri to the handler though. So perhaps have a `@classmethod`, `from_gcs_artifacts_uri` that we can call to instantiate the user's `Handler`? WDYT?

Reply:
I'm a bit confused. I think `Predictor` loads a model instead of `Handler`, doesn't it?

Or you mean a `@class method` `from_gcs_artifacts_uri` in `Predictor` as the following code?
```
class Handler:
    def __init__(self, gcs_artifacts_uri):
        self.predictor = CustomPredictor.from_gcs_artifacts_uri(gcs_artifacts_uri)
```
There's another problem here that we need to `import CustomPredictor` before users implement theirs. The current way to pass `predictor` to `Handler` avoids this issue.

We could consider marking `predictor: Optional[Predictor] = None` in `Handler.__init__` but `PredictionHandler` must have `predictor` passed in our design. Therefore, I didn't have strong preference to `Optional`.

Reply:
Updated the code. PTAL.